### PR TITLE
Add support for scalars from ResponseBody

### LINKED ISSUE #1459: Add converter for scalars to response body

### PR DESCRIPTION
**Pull request by JayNewstrom**
Closes #1459

### CODE REVIEW

**Comment by connorMathews** [comment]
Hey Jay, can you make the converters Singletons since they're stateless SAMs?

```
static final class ByteResponseBodyConverter implements Converter<ResponseBody, Byte> {
   static final ByteResponseBodyConverter INSTANCE = ByteResponseBodyConverter();
   
    @Override public Byte convert(ResponseBody value) throws IOException {
        return Byte.valueOf(value.string());
    }
}

...

if (type == Boolean.class) {
    return BooleanResponseBodyConverter.INSTANCE;
}
```

instead of 

```
static final class ByteResponseBodyConverter implements Converter<ResponseBody, Byte> {
    @Override public Byte convert(ResponseBody value) throws IOException {
        return Byte.valueOf(value.string());
    }
}

...

if (type == Boolean.class) {
    return new BooleanResponseBodyConverter();
}
```

It'll reduce object allocations.

Thanks!

**Comment by JayNewstrom** [comment]
That's how `ScalarRequestBodyConverter.INSTANCE` is done as well. Updated.

File: retrofit-converters/scalars/src/main/java/retrofit2/ScalarResponseBodyConverters.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package retrofit2;
```
Comment (JakeWharton) [inline retrofit-converters/scalars/src/main/java/retrofit2/ScalarResponseBodyConverters.java]: Copy license header from another file. Use 2016 for the date.

**Comment by JakeWharton** [comment]
Looks great! Just need a license header and we're good to go.

**Comment by JayNewstrom** [comment]
Whoops! Updated.

File: retrofit-converters/scalars/src/main/java/retrofit2/ScalarResponseBodyConverters.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    @Override public Byte convert(ResponseBody value) throws IOException {
+      return Byte.valueOf(value.string());
+    }
+  }
+
+  static final class CharacterResponseBodyConverter implements Converter<ResponseBody, Character> {
+    static final CharacterResponseBodyConverter INSTANCE = new CharacterResponseBodyConverter();
+
+    @Override public Character convert(ResponseBody value) throws IOException {
+      return value.string().charAt(0);
```
Comment (JakeWharton) [inline retrofit-converters/scalars/src/main/java/retrofit2/ScalarResponseBodyConverters.java]: I'm thinking that this should assert that the string is exactly 1 character and throw an `IOException` for a non-1 length. And a test case for both the 0 and non-1 lengths.

Comment (JakeWharton) [inline retrofit-converters/scalars/src/main/java/retrofit2/ScalarResponseBodyConverters.java]: Actually I'll just make this change locally during the merge.

Comment (JayNewstrom) [inline retrofit-converters/scalars/src/main/java/retrofit2/ScalarResponseBodyConverters.java]: :+1:

**Comment by JakeWharton** [comment]
Merged manually. Thanks!

**Pull request closed without merging**